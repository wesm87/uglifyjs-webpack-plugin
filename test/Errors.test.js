/* eslint-disable
  import/order
 */

import webpack from './helpers/compiler';
import { cleanErrorStack } from './helpers';
import UglifyJsPlugin from '../src/index';

describe('Errors', () => {
  test('ValidationError', async () => {
    const config = {
      plugins: [
        new UglifyJsPlugin({
          uglifyOptions: {
            output: {
              invalid: true,
            },
          },
        }),
      ],
    };

    const stats = await webpack('entry.js', config);
    const { assets } = stats.compilation;

    const errors = stats.compilation.errors.map(cleanErrorStack);
    const warnings = stats.compilation.warnings.map(cleanErrorStack);

    expect(errors).toMatchSnapshot();
    expect(warnings).toMatchSnapshot();

    for (const asset in assets) {
      if (assets[asset]) {
        expect(assets[asset].source()).toMatchSnapshot();
      }
    }
  });

  test('Uglify Errors', async () => {
    const config = {
      plugins: [
        new UglifyJsPlugin({
          uglifyOptions: {
            output: {
              invalid: true,
            },
          },
        }),
      ],
    };

    const stats = await webpack('entry.js', config);
    const { errors } = stats.compilation;

    expect(errors[0].message).toEqual(expect.stringContaining('from UglifyJs'));

    // const pluginEnvironment = new PluginEnvironment();
    // const compilerEnv = pluginEnvironment.getEnvironmentStub();
    // compilerEnv.context = '';

    // const plugin = new UglifyJsPlugin({
    //   uglifyOptions: {
    //     output: {
    //       'invalid-option': true,
    //     },
    //   },
    // });
    // plugin.apply(compilerEnv);
    // const [eventBinding] = pluginEnvironment.getEventBindings();

    // const chunkPluginEnvironment = new PluginEnvironment();
    // const compilation = chunkPluginEnvironment.getEnvironmentStub();
    // compilation.assets = {
    //   'test.js': {
    //     source: () => 'var foo = 1;',
    //   },
    // };
    // compilation.errors = [];

    // eventBinding.handler(compilation);
    // const [compilationEventBinding] = chunkPluginEnvironment.getEventBindings();

    // compilationEventBinding.handler(
    //   [
    //     {
    //       files: ['test.js'],
    //     },
    //   ],
    //   () => {
    //     expect(compilation.errors.length).toBe(1);
    //     expect(compilation.errors[0]).toBeInstanceOf(Error);
    //     expect(compilation.errors[0].message).toEqual(
    //       expect.stringContaining('from UglifyJs')
    //     );
    //   }
    // );
  });
});
