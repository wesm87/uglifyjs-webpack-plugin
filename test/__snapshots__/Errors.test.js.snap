// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors ValidationError 1`] = `
Array [
  "Error: main.bundle.js from UglifyJs
DefaultsError: \`invalid\` is not a supported option",
  "Error: runtime.bundle.js from UglifyJs
DefaultsError: \`invalid\` is not a supported option",
]
`;

exports[`Errors ValidationError 2`] = `
Array [
  "NoModeWarning: configuration
The 'mode' option has not been set. Set 'mode' option to 'development' or 'production' to enable defaults for this enviroment. ",
]
`;

exports[`Errors ValidationError 3`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports) {

// foo
/* @preserve*/
// bar
const a = 2 + 2;

module.exports = function Foo() {
  const b = 2 + 2;
  console.log(b + 1 + 2);
};


/***/ })
],[[0,1,0]]]);
//# sourceMappingURL=main.bundle.js.map"
`;

exports[`Errors ValidationError 4`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// install a JSONP callback for chunk loading
/******/ 	function webpackJsonpCallback(data) {
/******/ 		var chunkIds = data[0], moreModules = data[1], executeModules = data[2];
/******/ 		// add \\"moreModules\\" to the modules object,
/******/ 		// then flag all \\"chunkIds\\" as loaded and fire callback
/******/ 		var moduleId, chunkId, i = 0, resolves = [], result;
/******/ 		for(;i < chunkIds.length; i++) {
/******/ 			chunkId = chunkIds[i];
/******/ 			if(installedChunks[chunkId]) {
/******/ 				resolves.push(installedChunks[chunkId][0]);
/******/ 			}
/******/ 			installedChunks[chunkId] = 0;
/******/ 		}
/******/ 		for(moduleId in moreModules) {
/******/ 			if(Object.prototype.hasOwnProperty.call(moreModules, moduleId)) {
/******/ 				modules[moduleId] = moreModules[moduleId];
/******/ 			}
/******/ 		}
/******/ 		if(parentJsonpFunction) parentJsonpFunction(data);
/******/ 		while(resolves.length) {
/******/ 			resolves.shift()();
/******/ 		}
/******/ 		scheduledModules.push.apply(scheduledModules, executeModules || []);
/******/
/******/ 		for(i = 0; i < scheduledModules.length; i++) {
/******/ 			var scheduledModule = scheduledModules[i];
/******/ 			var fullfilled = true;
/******/ 			for(var j = 1; j < scheduledModule.length; j++) {
/******/ 				var depId = scheduledModule[j];
/******/ 				if(installedChunks[depId] !== 0) fullfilled = false;
/******/ 			}
/******/ 			if(fullfilled) {
/******/ 				scheduledModules.splice(i--, 1);
/******/ 				result = __webpack_require__(__webpack_require__.s = scheduledModule[0]);
/******/ 			}
/******/ 		}
/******/ 		return result;
/******/ 	};
/******/
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// object to store loaded and loading chunks
/******/ 	var installedChunks = {
/******/ 		1: 0
/******/ 	};
/******/
/******/ 	var scheduledModules = [];
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/ 	// This file contains only the entry chunk.
/******/ 	// The chunk loading function for additional chunks
/******/ 	__webpack_require__.e = function requireEnsure(chunkId) {
/******/ 		var promises = [];
/******/
/******/
/******/ 		// JSONP chunk loading for javascript
/******/
/******/ 		var installedChunkData = installedChunks[chunkId];
/******/ 		if(installedChunkData !== 0) { // 0 means \\"already installed\\".
/******/
/******/ 			// a Promise means \\"currently loading\\".
/******/ 			if(installedChunkData) {
/******/ 				promises.push(installedChunkData[2]);
/******/ 			} else {
/******/ 				// setup Promise in chunk cache
/******/ 				var promise = new Promise(function(resolve, reject) {
/******/ 					installedChunkData = installedChunks[chunkId] = [resolve, reject];
/******/ 				});
/******/ 				promises.push(installedChunkData[2] = promise);
/******/
/******/ 				// start chunk loading
/******/ 				var head = document.getElementsByTagName('head')[0];
/******/ 				var script = document.createElement('script');
/******/ 				script.charset = 'utf-8';
/******/ 				script.timeout = 120000;
/******/
/******/ 				if (__webpack_require__.nc) {
/******/ 					script.setAttribute(\\"nonce\\", __webpack_require__.nc);
/******/ 				}
/******/ 				script.src = __webpack_require__.p + \\"\\" + chunkId + \\".bundle.js\\";
/******/ 				var timeout = setTimeout(function(){
/******/ 					onScriptComplete({ type: 'timeout', target: script });
/******/ 				}, 120000);
/******/ 				script.onerror = script.onload = onScriptComplete;
/******/ 				function onScriptComplete(event) {
/******/ 					// avoid mem leaks in IE.
/******/ 					script.onerror = script.onload = null;
/******/ 					clearTimeout(timeout);
/******/ 					var chunk = installedChunks[chunkId];
/******/ 					if(chunk !== 0) {
/******/ 						if(chunk) {
/******/ 							var errorType = event && (event.type === 'load' ? 'missing' : event.type);
/******/ 							var realSrc = event && event.target && event.target.src;
/******/ 							var error = new Error('Loading chunk ' + chunkId + ' failed.\\\\n(' + errorType + ': ' + realSrc + ')');
/******/ 							error.type = errorType;
/******/ 							error.request = realSrc;
/******/ 							chunk[1](error);
/******/ 						}
/******/ 						installedChunks[chunkId] = undefined;
/******/ 					}
/******/ 				};
/******/ 				head.appendChild(script);
/******/ 			}
/******/ 		}
/******/ 		return Promise.all(promises);
/******/ 	};
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, {
/******/ 				configurable: false,
/******/ 				enumerable: true,
/******/ 				get: getter
/******/ 			});
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/ 	// on error function for async loading
/******/ 	__webpack_require__.oe = function(err) { console.error(err); throw err; };
/******/
/******/ 	var jsonpArray = window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || [];
/******/ 	var parentJsonpFunction = jsonpArray.push.bind(jsonpArray);
/******/ 	jsonpArray.push = webpackJsonpCallback;
/******/ 	jsonpArray = jsonpArray.slice();
/******/ 	for(var i = 0; i < jsonpArray.length; i++) webpackJsonpCallback(jsonpArray[i]);
/******/ })
/************************************************************************/
/******/ ([]);
//# sourceMappingURL=runtime.bundle.js.map"
`;

exports[`Errors ValidationError 5`] = `"{\\"version\\":3,\\"sources\\":[\\"webpack:///./entry.js\\"],\\"names\\":[],\\"mappings\\":\\";;;;AAAA;AACA;AACA;AACA;;AAEA;AACA;AACA;AACA\\",\\"file\\":\\"main.bundle.js\\",\\"sourcesContent\\":[\\"// foo\\\\n/* @preserve*/\\\\n// bar\\\\nconst a = 2 + 2;\\\\n\\\\nmodule.exports = function Foo() {\\\\n  const b = 2 + 2;\\\\n  console.log(b + 1 + 2);\\\\n};\\\\n\\\\n\\\\n\\\\n//////////////////\\\\n// WEBPACK FOOTER\\\\n// ./entry.js\\\\n// module id = 0\\\\n// module chunks = 0\\"],\\"sourceRoot\\":\\"\\"}"`;

exports[`Errors ValidationError 6`] = `"{\\"version\\":3,\\"sources\\":[\\"webpack:///webpack/bootstrap\\"],\\"names\\":[],\\"mappings\\":\\";AAAA;AACA;AACA;AACA;AACA;AACA;AACA,gBAAQ,oBAAoB;AAC5B;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;;AAEA,qBAAa,6BAA6B;AAC1C;AACA;AACA,0BAAkB,4BAA4B;AAC9C;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;;AAEA;AACA;;AAEA;AACA;AACA;AACA;;AAEA;;AAEA;AACA;;AAEA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;;AAEA;AACA;;AAEA;AACA;;AAEA;AACA;AACA;;AAEA;AACA;AACA;AACA;;;AAGA;;AAEA;AACA,yCAAiC;;AAEjC;AACA;AACA;AACA,aAAK;AACL;AACA;AACA;AACA,cAAM;AACN;;AAEA;AACA;AACA;AACA;AACA;;AAEA;AACA;AACA;AACA;AACA;AACA,gCAAwB,kCAAkC;AAC1D,cAAM;AACN;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;;AAEA;AACA;;AAEA;AACA;;AAEA;AACA;AACA;AACA;AACA;AACA;AACA;AACA,aAAK;AACL;AACA;;AAEA;AACA;AACA,yDAAiD,cAAc;AAC/D;;AAEA;AACA;AACA;AACA,mCAA2B,0BAA0B,EAAE;AACvD,yCAAiC,eAAe;AAChD;AACA;AACA;;AAEA;AACA,8DAAsD,+DAA+D;;AAErH;AACA;;AAEA;AACA,kDAA0C,oBAAoB,WAAW;;AAEzE;AACA;AACA;AACA;AACA,wBAAgB,uBAAuB\\",\\"file\\":\\"runtime.bundle.js\\",\\"sourcesContent\\":[\\" \\\\t// install a JSONP callback for chunk loading\\\\n \\\\tfunction webpackJsonpCallback(data) {\\\\n \\\\t\\\\tvar chunkIds = data[0], moreModules = data[1], executeModules = data[2];\\\\n \\\\t\\\\t// add \\\\\\"moreModules\\\\\\" to the modules object,\\\\n \\\\t\\\\t// then flag all \\\\\\"chunkIds\\\\\\" as loaded and fire callback\\\\n \\\\t\\\\tvar moduleId, chunkId, i = 0, resolves = [], result;\\\\n \\\\t\\\\tfor(;i < chunkIds.length; i++) {\\\\n \\\\t\\\\t\\\\tchunkId = chunkIds[i];\\\\n \\\\t\\\\t\\\\tif(installedChunks[chunkId]) {\\\\n \\\\t\\\\t\\\\t\\\\tresolves.push(installedChunks[chunkId][0]);\\\\n \\\\t\\\\t\\\\t}\\\\n \\\\t\\\\t\\\\tinstalledChunks[chunkId] = 0;\\\\n \\\\t\\\\t}\\\\n \\\\t\\\\tfor(moduleId in moreModules) {\\\\n \\\\t\\\\t\\\\tif(Object.prototype.hasOwnProperty.call(moreModules, moduleId)) {\\\\n \\\\t\\\\t\\\\t\\\\tmodules[moduleId] = moreModules[moduleId];\\\\n \\\\t\\\\t\\\\t}\\\\n \\\\t\\\\t}\\\\n \\\\t\\\\tif(parentJsonpFunction) parentJsonpFunction(data);\\\\n \\\\t\\\\twhile(resolves.length) {\\\\n \\\\t\\\\t\\\\tresolves.shift()();\\\\n \\\\t\\\\t}\\\\n \\\\t\\\\tscheduledModules.push.apply(scheduledModules, executeModules || []);\\\\n\\\\n \\\\t\\\\tfor(i = 0; i < scheduledModules.length; i++) {\\\\n \\\\t\\\\t\\\\tvar scheduledModule = scheduledModules[i];\\\\n \\\\t\\\\t\\\\tvar fullfilled = true;\\\\n \\\\t\\\\t\\\\tfor(var j = 1; j < scheduledModule.length; j++) {\\\\n \\\\t\\\\t\\\\t\\\\tvar depId = scheduledModule[j];\\\\n \\\\t\\\\t\\\\t\\\\tif(installedChunks[depId] !== 0) fullfilled = false;\\\\n \\\\t\\\\t\\\\t}\\\\n \\\\t\\\\t\\\\tif(fullfilled) {\\\\n \\\\t\\\\t\\\\t\\\\tscheduledModules.splice(i--, 1);\\\\n \\\\t\\\\t\\\\t\\\\tresult = __webpack_require__(__webpack_require__.s = scheduledModule[0]);\\\\n \\\\t\\\\t\\\\t}\\\\n \\\\t\\\\t}\\\\n \\\\t\\\\treturn result;\\\\n \\\\t};\\\\n\\\\n \\\\t// The module cache\\\\n \\\\tvar installedModules = {};\\\\n\\\\n \\\\t// object to store loaded and loading chunks\\\\n \\\\tvar installedChunks = {\\\\n \\\\t\\\\t1: 0\\\\n \\\\t};\\\\n\\\\n \\\\tvar scheduledModules = [];\\\\n\\\\n \\\\t// The require function\\\\n \\\\tfunction __webpack_require__(moduleId) {\\\\n\\\\n \\\\t\\\\t// Check if module is in cache\\\\n \\\\t\\\\tif(installedModules[moduleId]) {\\\\n \\\\t\\\\t\\\\treturn installedModules[moduleId].exports;\\\\n \\\\t\\\\t}\\\\n \\\\t\\\\t// Create a new module (and put it into the cache)\\\\n \\\\t\\\\tvar module = installedModules[moduleId] = {\\\\n \\\\t\\\\t\\\\ti: moduleId,\\\\n \\\\t\\\\t\\\\tl: false,\\\\n \\\\t\\\\t\\\\texports: {}\\\\n \\\\t\\\\t};\\\\n\\\\n \\\\t\\\\t// Execute the module function\\\\n \\\\t\\\\tmodules[moduleId].call(module.exports, module, module.exports, __webpack_require__);\\\\n\\\\n \\\\t\\\\t// Flag the module as loaded\\\\n \\\\t\\\\tmodule.l = true;\\\\n\\\\n \\\\t\\\\t// Return the exports of the module\\\\n \\\\t\\\\treturn module.exports;\\\\n \\\\t}\\\\n\\\\n \\\\t// This file contains only the entry chunk.\\\\n \\\\t// The chunk loading function for additional chunks\\\\n \\\\t__webpack_require__.e = function requireEnsure(chunkId) {\\\\n \\\\t\\\\tvar promises = [];\\\\n\\\\n\\\\n \\\\t\\\\t// JSONP chunk loading for javascript\\\\n\\\\n \\\\t\\\\tvar installedChunkData = installedChunks[chunkId];\\\\n \\\\t\\\\tif(installedChunkData !== 0) { // 0 means \\\\\\"already installed\\\\\\".\\\\n\\\\n \\\\t\\\\t\\\\t// a Promise means \\\\\\"currently loading\\\\\\".\\\\n \\\\t\\\\t\\\\tif(installedChunkData) {\\\\n \\\\t\\\\t\\\\t\\\\tpromises.push(installedChunkData[2]);\\\\n \\\\t\\\\t\\\\t} else {\\\\n \\\\t\\\\t\\\\t\\\\t// setup Promise in chunk cache\\\\n \\\\t\\\\t\\\\t\\\\tvar promise = new Promise(function(resolve, reject) {\\\\n \\\\t\\\\t\\\\t\\\\t\\\\tinstalledChunkData = installedChunks[chunkId] = [resolve, reject];\\\\n \\\\t\\\\t\\\\t\\\\t});\\\\n \\\\t\\\\t\\\\t\\\\tpromises.push(installedChunkData[2] = promise);\\\\n\\\\n \\\\t\\\\t\\\\t\\\\t// start chunk loading\\\\n \\\\t\\\\t\\\\t\\\\tvar head = document.getElementsByTagName('head')[0];\\\\n \\\\t\\\\t\\\\t\\\\tvar script = document.createElement('script');\\\\n \\\\t\\\\t\\\\t\\\\tscript.charset = 'utf-8';\\\\n \\\\t\\\\t\\\\t\\\\tscript.timeout = 120000;\\\\n\\\\n \\\\t\\\\t\\\\t\\\\tif (__webpack_require__.nc) {\\\\n \\\\t\\\\t\\\\t\\\\t\\\\tscript.setAttribute(\\\\\\"nonce\\\\\\", __webpack_require__.nc);\\\\n \\\\t\\\\t\\\\t\\\\t}\\\\n \\\\t\\\\t\\\\t\\\\tscript.src = __webpack_require__.p + \\\\\\"\\\\\\" + chunkId + \\\\\\".bundle.js\\\\\\";\\\\n \\\\t\\\\t\\\\t\\\\tvar timeout = setTimeout(function(){\\\\n \\\\t\\\\t\\\\t\\\\t\\\\tonScriptComplete({ type: 'timeout', target: script });\\\\n \\\\t\\\\t\\\\t\\\\t}, 120000);\\\\n \\\\t\\\\t\\\\t\\\\tscript.onerror = script.onload = onScriptComplete;\\\\n \\\\t\\\\t\\\\t\\\\tfunction onScriptComplete(event) {\\\\n \\\\t\\\\t\\\\t\\\\t\\\\t// avoid mem leaks in IE.\\\\n \\\\t\\\\t\\\\t\\\\t\\\\tscript.onerror = script.onload = null;\\\\n \\\\t\\\\t\\\\t\\\\t\\\\tclearTimeout(timeout);\\\\n \\\\t\\\\t\\\\t\\\\t\\\\tvar chunk = installedChunks[chunkId];\\\\n \\\\t\\\\t\\\\t\\\\t\\\\tif(chunk !== 0) {\\\\n \\\\t\\\\t\\\\t\\\\t\\\\t\\\\tif(chunk) {\\\\n \\\\t\\\\t\\\\t\\\\t\\\\t\\\\t\\\\tvar errorType = event && (event.type === 'load' ? 'missing' : event.type);\\\\n \\\\t\\\\t\\\\t\\\\t\\\\t\\\\t\\\\tvar realSrc = event && event.target && event.target.src;\\\\n \\\\t\\\\t\\\\t\\\\t\\\\t\\\\t\\\\tvar error = new Error('Loading chunk ' + chunkId + ' failed.\\\\\\\\n(' + errorType + ': ' + realSrc + ')');\\\\n \\\\t\\\\t\\\\t\\\\t\\\\t\\\\t\\\\terror.type = errorType;\\\\n \\\\t\\\\t\\\\t\\\\t\\\\t\\\\t\\\\terror.request = realSrc;\\\\n \\\\t\\\\t\\\\t\\\\t\\\\t\\\\t\\\\tchunk[1](error);\\\\n \\\\t\\\\t\\\\t\\\\t\\\\t\\\\t}\\\\n \\\\t\\\\t\\\\t\\\\t\\\\t\\\\tinstalledChunks[chunkId] = undefined;\\\\n \\\\t\\\\t\\\\t\\\\t\\\\t}\\\\n \\\\t\\\\t\\\\t\\\\t};\\\\n \\\\t\\\\t\\\\t\\\\thead.appendChild(script);\\\\n \\\\t\\\\t\\\\t}\\\\n \\\\t\\\\t}\\\\n \\\\t\\\\treturn Promise.all(promises);\\\\n \\\\t};\\\\n\\\\n \\\\t// expose the modules object (__webpack_modules__)\\\\n \\\\t__webpack_require__.m = modules;\\\\n\\\\n \\\\t// expose the module cache\\\\n \\\\t__webpack_require__.c = installedModules;\\\\n\\\\n \\\\t// define getter function for harmony exports\\\\n \\\\t__webpack_require__.d = function(exports, name, getter) {\\\\n \\\\t\\\\tif(!__webpack_require__.o(exports, name)) {\\\\n \\\\t\\\\t\\\\tObject.defineProperty(exports, name, {\\\\n \\\\t\\\\t\\\\t\\\\tconfigurable: false,\\\\n \\\\t\\\\t\\\\t\\\\tenumerable: true,\\\\n \\\\t\\\\t\\\\t\\\\tget: getter\\\\n \\\\t\\\\t\\\\t});\\\\n \\\\t\\\\t}\\\\n \\\\t};\\\\n\\\\n \\\\t// define __esModule on exports\\\\n \\\\t__webpack_require__.r = function(exports) {\\\\n \\\\t\\\\tObject.defineProperty(exports, '__esModule', { value: true });\\\\n \\\\t};\\\\n\\\\n \\\\t// getDefaultExport function for compatibility with non-harmony modules\\\\n \\\\t__webpack_require__.n = function(module) {\\\\n \\\\t\\\\tvar getter = module && module.__esModule ?\\\\n \\\\t\\\\t\\\\tfunction getDefault() { return module['default']; } :\\\\n \\\\t\\\\t\\\\tfunction getModuleExports() { return module; };\\\\n \\\\t\\\\t__webpack_require__.d(getter, 'a', getter);\\\\n \\\\t\\\\treturn getter;\\\\n \\\\t};\\\\n\\\\n \\\\t// Object.prototype.hasOwnProperty.call\\\\n \\\\t__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };\\\\n\\\\n \\\\t// __webpack_public_path__\\\\n \\\\t__webpack_require__.p = \\\\\\"\\\\\\";\\\\n\\\\n \\\\t// on error function for async loading\\\\n \\\\t__webpack_require__.oe = function(err) { console.error(err); throw err; };\\\\n\\\\n \\\\tvar jsonpArray = window[\\\\\\"webpackJsonp\\\\\\"] = window[\\\\\\"webpackJsonp\\\\\\"] || [];\\\\n \\\\tvar parentJsonpFunction = jsonpArray.push.bind(jsonpArray);\\\\n \\\\tjsonpArray.push = webpackJsonpCallback;\\\\n \\\\tjsonpArray = jsonpArray.slice();\\\\n \\\\tfor(var i = 0; i < jsonpArray.length; i++) webpackJsonpCallback(jsonpArray[i]);\\\\n\\\\n\\\\n\\\\n// WEBPACK FOOTER //\\\\n// webpack/bootstrap\\"],\\"sourceRoot\\":\\"\\"}"`;
